### PR TITLE
ADEN-8010 Remaining tests

### PR DESCRIPTION
**Pull request by emadry**

### CODE REVIEW

File: tests/common/helpers.js [hunk 1]
```diff
@@ -120,6 +120,10 @@ class Helpers {
 		return browser.element(adSlot).getAttribute(adSlots.lineItemIdAttribute);
 	}
 
+	isLineItemExisitng(adSlot) {
```
Comment (jacekbj) [inline tests/common/helpers.js]: typo `isLineItemExisitng` -> `isLineItemExisting`

File: tests/common/query-strings.js [hunk 2]
```diff
@@ -121,13 +137,59 @@ class QueryStrings {
...
+		const on = enabled ? '1' : '0';
+
+		return `${this.bidders.deals}=${on}`;
+	}
+
 	getSessionIdParam(parameter) {
 		return `${this.utils.sessionId}=${parameter}`;
 	}
 
+	getProjects(...projects) {
```
Comment (jacekbj) [inline tests/common/query-strings.js]: Not used?

File: tests/common/query-strings.js [hunk 3]
```diff
@@ -121,13 +137,59 @@ class QueryStrings {
...
+		return `${this.services.adEngineDelay}=${timeout}`;
+	}
+
+	getEmptyResponse(empty) {
+		const on = empty ? '1' : '0';
+
+		return `${this.utils.forceEmptyResponse}=${on}`;
+	}
+
+	getDeals(enabled) {
```
Comment (jacekbj) [inline tests/common/query-strings.js]: Not used?

File: tests/common/query-strings.js [hunk 4]
```diff
@@ -121,13 +137,59 @@ class QueryStrings {
...
 	getSessionIdParam(parameter) {
 		return `${this.utils.sessionId}=${parameter}`;
 	}
 
+	getProjects(...projects) {
+		return `${this.services.enabledProjects}=${projects.join(',')}`;
+	}
+
+	getTurnedOffSlots(...slots) {
+		return `${this.services.enabledProjects}=${slots.join(',')}`;
```
Comment (jacekbj) [inline tests/common/query-strings.js]: Funny that query string "enabled-project" is disables slots. Is it correct?

File: tests/pages/bill-the-lizard.page.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		this.projects = {
+			queenOfHears: 'queen_of_hearts',
+			cheshireCat: 'cheshirecat',
+		};
+
+		this.lazyCallCheshireCatButton = '#lazyCallCat';
+		this.lazyCallCheshireCatWithIdButton = '#lazyCallCatWithId';
+	}
+
+	getLastCallStatus() {
```
Comment (jacekbj) [inline tests/pages/bill-the-lizard.page.js]: These 4 methods are here for future use? They're not called anywhere.

File: tests/specs/bidders/a9.desktop.test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { expect } from 'chai';
+import { asserts } from '../../common/asserts';
+import { adSlots } from '../../common/ad-slots';
+import { helpers } from '../../common/helpers';
+import { a9 } from '../../pages/a9.page';
+import { queryStrings } from '../../common/query-strings';
+
+describe('A9', () => {
```
Comment (jacekbj) [inline tests/specs/bidders/a9.desktop.test.js]: This test is a 💎

File: tests/specs/services/moat.desktop.test.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import { moat } from '../../pages/moat.page';
+import { timeouts } from '../../common/timeouts';
+import { helpers } from '../../common/helpers';
+import { queryStrings } from '../../common/query-strings';
+import { adSlots } from '../../common/ad-slots';
+
+describe('It will test krux page', () => {
```
Comment (jacekbj) [inline tests/specs/services/moat.desktop.test.js]: `s/krux/moat`

File: tests/specs/services/moat.desktop.test.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	it('will test if cached value is stored', () => {
+		browser.url(moat.pageLink);
+		browser.waitForVisible(adSlots.topLeaderboard, timeouts.standard);
+		helpers.waitForValuesLoaded();
+		expect(moat.getPageLevelParams()).to.equal(
+			'{"m_safety":"safe","m_categories":["moat_safe"],"m_data":"0"}',
```
Comment (jacekbj) [inline tests/specs/services/moat.desktop.test.js]: If this test turns out to be flaky we may need to load page level params from JSON and make a deep comparison.

File: tests/specs/services/moat.desktop.test.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+		helpers.waitForViewabillityCounted();
+		expect(moat.getPageLevelParams()).to.include('Waiting');
+	});
+
+	it('will test delayed moat', () => {
+		helpers.navigateToUrl(moat.pageLink, queryStrings.getAdEngineDelay(1000));
+		browser.waitForVisible(adSlots.topLeaderboard, timeouts.standard);
+		helpers.waitForViewabillityCounted();
+		helpers.waitForValuesLoaded();
+		expect(moat.getPageLevelParams()).to.equal(
```
Comment (jacekbj) [inline tests/specs/services/moat.desktop.test.js]: ditto

File: tests/specs/templates/outstream.desktop.test.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
+/**/ import { expect } from 'chai';
```
Comment (jacekbj) [inline tests/specs/templates/outstream.desktop.test.js]: unnecessary comment

File: tests/specs/templates/sticky-tlb.desktop.test.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
+import { expect } from 'chai';
+import { stickyTlb } from '../../pages/sticky-tlb.page';
+import { adSlots } from '../../common/ad-slots';
+import { timeouts } from '../../common/timeouts';
+import { helpers } from '../../common/helpers';
+import { queryStrings } from '../../common/query-strings';
+
+describe('sticky-ad template', () => {
```
Comment (jacekbj) [inline tests/specs/templates/sticky-tlb.desktop.test.js]: `s/ad/tlb`

File: tests/common/query-strings.js [hunk 12]
```diff
@@ -121,13 +137,59 @@ class QueryStrings {
...
+	}
+
 	constructInstantGlobal(wg, country = 'XX', percent = null, additional = null) {
 		return `${wg}=${this.getIGParameters(country, percent, additional)}`;
 	}
+
+	constructSingleGeoInstantGlobal(country = 'XX', percent = null) {
+		return `${this.utils.enabledGeo}=${country}/${percent}`;
```
Comment (jacekbj) [inline tests/common/query-strings.js]: String `enabled-geo` is not used anywhere. Is this method correct?

File: tests/common/query-strings.js (hunk 2 above)
Comment (emadry) [reply tests/common/query-strings.js]: Yes. Some methods were prepared for future tests.

File: tests/common/query-strings.js (hunk 12 above)
Comment (emadry) [reply tests/common/query-strings.js]: It's used in sticky-tlb tests and it's working

**Pull request merged**